tests/orquestra/quantum/circuits/_gates_test.py

```diff
@@ -213,6 +220,77 @@ def test_constructing_controlled_gate_with_zero_control_raises_error(self, gate)
             gate.controlled(0)
 
 
+@pytest.mark.parametrize("gate", GATES_REPRESENTATIVES)
+class TestDagger:
+    def test_power_of_dagger(self, gate):
+        if len(gate.free_symbols) == 0:
+            gate.dagger.power(0.5)
```

Comment:
Maybe I'm missing something, but why is this its own class? We have several tests of dagger above. Maybe we should either put this test up there or take all the dagger tests and put them here?

Reply:
I put that test in it's own class as I didn't see a test class dedicated to testing `Dagger` and wasn't aware that `TestMatrixFactoryGate` was playing that role. I'm my opinion I think that it would be best to have tests specific to `Dagger` and `MatrixFactoryGate` in their own classes respectively. That would allow us to easily find tests specific to a class and also have more certainty that a hypothetical test failure is related to the implementation `Dagger` rather than that of `MatrixFactoryGate`.

For the time being, I've removed the `TestDagger` class and moved the new `Dagger` test into `TestMatrixFactoryGate` with the others. If you agree with refactoring those tests in the way I proposed, I think it would be good to apply it in a separate pull request to keep the scope of this one well defined.